addons/product/models/product.py

```diff
@@ -388,17 +388,36 @@ def _name_get(d):
         self.check_access_rule("read")
 
         result = []
+
+        # Prefetch the fields used by the `name_get`, so `browse` doesn't fetch other fields
+        # Use `load=False` to not call `name_get` for the `product_tmpl_id`
+        self.sudo().read(['name', 'default_code', 'product_tmpl_id', 'attribute_value_ids'], load=False)
+
+        product_template_ids = [product.product_tmpl_id.id for product in self]
+
+        if partner_ids:
+            supplier_info = self.env['product.supplierinfo'].sudo().search([
+                ('product_tmpl_id', 'in', product_template_ids),
+                ('name', 'in', partner_ids),
+            ])
+            # Prefetch the fields used by the `name_get`, so `browse` doesn't fetch other fields
+            # Use `load=False` to not call `name_get` for the `product_tmpl_id` and `product_id`
+            supplier_info.sudo().read(['product_tmpl_id', 'product_id', 'product_name', 'product_code'], load=False)
```

Comment:
@beledouxdenis This line will be a nightmare in a multi-company context as you will return all records from all companies

Reply:
I don't see the difference compared to beefore:
```
for product in self.sudo():
...
 sellers = [x for x in product.seller_ids if (x.name.id in partner_ids) and (x.product_id == product)]
```
The sellers are therefore read as sudo as well.

Am I missing something ?

Reply:
Hi @beledouxdenis . You did not miss something but it was already ugly before ;-)